checker/src/main/java/alluxio/checker/SparkIntegrationChecker.java

```diff
@@ -0,0 +1,265 @@
+      }
+    }
+
+    // Runs a Spark job to check Spark executors configuration
+    return runOperations(sc);
+  }
+
+  /**
+   * Spark job to check whether Spark executors can recognize Alluxio filesystem.
+   *
+   * @param sc current JavaSparkContext
+   * @return 0 on success, 1 on failures
+   */
+  private static int runOperations(JavaSparkContext sc) {
+    // Generates a list of integer for testing
+    List<Integer> nums = IntStream.rangeClosed(1, sPartition).boxed().collect(Collectors.toList());
+
+    JavaRDD<Integer> dataSet = sc.parallelize(nums, sPartition);
+
+    // Runs a Spark job to check whether Spark executors can recognize Alluxio
+    JavaPairRDD<Integer, String> extractedStatus = dataSet
+        .mapToPair(s -> new Tuple2<>(getStatus(), getAddress(s)));
+
+    // Merges the IP addresses that can/cannot recognize Alluxio
+    JavaPairRDD<Integer, String> mergeStatus = extractedStatus.reduceByKey((a, b) -> merge(a, b));
+
+    sSparkJobResult = mergeStatus.collect();
+
+    // If one executor cannot recognize Alluxio, something wrong has happened
+    boolean canRecognizeClass = true;
+    boolean canRecognizeFS = true;
+    for (Tuple2<Integer, String> op : sSparkJobResult) {
+      if (op._1() == 1) {
+        canRecognizeClass = false;
+      }
+      if (op._1() == 2) {
+        canRecognizeFS = false;
+      }
+    }
+    return canRecognizeClass ? (canRecognizeFS ? 0 : 2) : 1;
+  }
+
+  /**
+   * Checks if this Spark driver or executors can recognize Alluxio classes and filesystem.
+   *
+   * @return 0 succeed, 1 fail to recognize Alluxio classes,
+   * 2 fail to recognize Alluxio filesystem
+   */
+  private static int getStatus() {
+    // Checks if Spark driver or executors can recognize Alluxio classes
+    try {
+      Class.forName("alluxio.AlluxioURI");
+      Class.forName("alluxio.client.file.BaseFileSystem");
+      Class.forName("alluxio.hadoop.AlluxioFileSystem");
+    } catch (ClassNotFoundException e) {
+      return 1;
+    }
+
+    // checks if Spark driver or executors can recognize Alluxio filesystem
+    try {
+      FileSystem.getFileSystemClass("alluxio", new Configuration());
+    } catch (Exception e) {
+      return 2;
+    }
+
+    return 0;
+  }
+
+  /**
+   * Gets the current Spark executor IP address.
+   *
+   * @param s the integer pass in
+   * @return the current Spark executor IP address
+   */
+  private static String getAddress(Integer s) {
+    String address;
+    try {
+      address = InetAddress.getLocalHost().getHostAddress();
+    } catch (UnknownHostException e) {
+      LOG.debug("cannot get host address");
+      address = "unknown address";
+    }
+    return address;
+  }
+
+  /**
+   * Merges the IP addresses that have the same key.
+   *
+   * @param a, b two IP addresses that both can recognize Alluxio or both cannot
+   * @return merged Spark executor IP addresses
+   */
+  private static String merge(String a, String b) {
+    if (a.contains(b)) {
```

Comment:
should this be an equality check instead?

Reply:
reduceByKey merges all the IP addresses of the same Status, so I think a and b may not be a single ip.
a may be ip1; ip2; ip3. b may be ip3; ip4; ip5.